src/main/java/com/finalmarket/demo/logic/PredefinedLogicController.java

```diff
@@ -0,0 +1,17 @@
+package com.finalmarket.demo.logic;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class PredefinedLogicController {
+    @Autowired
+    private PredefinedLogicService service;
```

Comment:
I would rather put all available services here in the controller. And add scheduled method inside each of the services.